client/jetpack-onboarding/steps/contact-form.jsx

```diff
@@ -65,45 +63,32 @@ class JetpackOnboardingContactFormStep extends React.PureComponent {
 	}
 
 	renderActionTile() {
-		const { hasContactForm, isConnected, siteUrl, translate } = this.props;
-		const headerText = translate( "Let's grow your audience with Jetpack." );
-		const subHeaderText = (
-			<Fragment>
-				{ translate(
-					'A great first step is adding a Contact Us page that includes a contact form.'
-				) }
-				<br />
-				{ translate( 'Create a Jetpack account to unlock this and dozens of other features.' ) }
-			</Fragment>
-		);
-		const connectUrl = addQueryArgs(
-			{
-				url: siteUrl,
-				// TODO: add a parameter to the JPC to redirect back to this step after completion
-				// and in the redirect URL include the ?action=add_contact_form parameter
-				// to actually trigger the page and form creation action after getting back to JPO
-			},
-			'/jetpack/connect'
+		const { hasContactForm, siteId, translate } = this.props;
+		const header = (
+			<FormattedHeader
+				headerText={ translate( "Let's grow your audience with Jetpack." ) }
+				subHeaderText={
+					<Fragment>
+						{ translate(
+							'A great first step is adding a Contact Us page that includes a contact form.'
+						) }
+						<br />
+						{ translate( 'Create a Jetpack account to unlock this and dozens of other features.' ) }
+					</Fragment>
+				}
+			/>
 		);
-		const href = ! isConnected ? connectUrl : null;
 
 		return (
-			<Fragment>
-				<FormattedHeader headerText={ headerText } subHeaderText={ subHeaderText } />
-
-				<TileGrid>
-					<Tile
-						buttonLabel={ ! hasContactForm ? translate( 'Add a contact form' ) : null }
-						description={
-							hasContactForm ? translate( 'Your contact form has been created.' ) : null
-						}
-						image={ '/calypso/images/illustrations/contact-us.svg' }
-						onClick={ this.handleAddContactForm }
-						href={ href }
-						e2eType={ 'contact-form' }
-					/>
-				</TileGrid>
-			</Fragment>
+			<ConnectIntro
+				buttonLabel={ ! hasContactForm ? translate( 'Add a contact form' ) : null }
+				description={ hasContactForm ? translate( 'Your contact form has been created.' ) : null }
```

Comment:
Can we avoid the `? : null` ternaries by using `hasContactForm && ...` instead? The components that `ConnectIntro` use might be smart enough to correctly deal with a `false` here, no?

Reply:
We have a `PropTypes.string` for the description (which IMHO makes sense), so it would error out if we pass it a boolean value. I suggest that we keep it as-is.

Reply:
Yeah, I was expecting it might mess up our `propTypes` otherwise. 🙁  Let's keep it then.